haiku/_src/stateful.py

```diff
@@ -190,10 +186,7 @@ def wrapper(*args, **kwargs):
     kwargs["hk_state"] = internal_state()
     (value, (aux, hk_state)), grads = grad_fun(*args, **kwargs)
     update_internal_state(hk_state)
-    if has_aux:
-      return (value, aux), grads
-    else:
-      return value, grads
+    return (value, aux) if has_aux else value, grads
```

Comment:
This is hard for me to parse - I initially misread it as returning `(value, aux)` else `(value, grads)`.
Acknowledging that the previous version is much more verbose, I'd rather have verbose code than hard-to-parse code. Consider refactoring again or reverting this chunk of the change.